### PR TITLE
Fix: ignore unmergable imports when checking no-duplicate-imports (fixes #13180)

### PR DESCRIPTION
**Pull request by cmal**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [X] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[X] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fix the issue here: https://github.com/eslint/eslint/issues/12758

#### Is there anything you'd like reviewers to focus on?
1. I thought two nodes are mergable only when one is ImportNamespaceSpecifier and one is ImportSpecifier, is that true?
2. I thought only in handleImports when checking importsInFile should I change the `checkAndReport` function, is that true?

### CODE REVIEW

File: docs/rules/no-duplicate-imports.md [hunk 1]
```diff
@@ -33,6 +35,27 @@ import { merge, find } from 'module';
 import something from 'another-module';
 ```
 
+The following examples explains ***can be merged** in detail:
```
Comment (nzakas) [inline docs/rules/no-duplicate-imports.md:38]:
```suggestion
The following examples explains **can be merged** in detail:
```

An extra asterisk snuck in there

**Comment by nzakas** [comment]
Sorry, looks like we missed this PR! @mdjermanovic you reopened the issue related to this PR, can you take a look?

**Comment by cmal** [comment]
👍

File: lib/rules/no-duplicate-imports.js [hunk 2]
```diff
@@ -21,18 +21,107 @@ function getValue(node) {
...
+ *
+ * @returns {boolean} Whether or not the import's type is type.
+ */
+function checkImportType(node, type) {
+    return (
+        node.specifiers &&
+        node.specifiers[0] &&
+        node.specifiers[0].type === type
+    );
+}
```
Comment (mdjermanovic) [inline lib/rules/no-duplicate-imports.js:37]: We shouldn't check only the first specifier. For example, this shouldn't be reported:

```js
/* eslint no-duplicate-imports: error */

import foo, * as bar from 'mod';

import { baz } from 'mod';
```

File: lib/rules/no-duplicate-imports.js [hunk 3]
```diff
@@ -21,18 +21,107 @@ function getValue(node) {
...
+        (
+            checkImportType(node1, "ImportNamespaceSpecifier") &&
+            checkImportType(node2, "ImportSpecifier")
+        ) ||
+        (
+            checkImportType(node1, "ImportSpecifier") &&
+            checkImportType(node2, "ImportNamespaceSpecifier")
+        )
+    );
+}
```
Comment (mdjermanovic) [inline lib/rules/no-duplicate-imports.js:70]: Looks like this function is actually doing the opposite of its name: checks if two nodes _cannot_ be merged?

**Comment by nzakas** [comment]
@cmal there are a couple of changes requested. Can you please take a look?

**Comment by cmal** [comment]
OK, sorry for late. I'll have a look.

**Comment by mdjermanovic** [comment]
Closing due to inactivity. New PR: https://github.com/eslint/eslint/pull/14238

**Pull request closed without merging**